src/dotnet-user-secrets/Internal/InitCommand.cs

```diff
@@ -0,0 +1,99 @@
+using Microsoft.Extensions.CommandLineUtils;
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    public class InitCommand : ICommand
+    {
+        public string OverrideId { get; }
+        public string ProjectPath { get; }
+        public string WorkingDirectory { get; private set; } = Directory.GetCurrentDirectory();
+
+        internal static void Configure(CommandLineApplication command, CommandLineOptions options)
+        {
+            command.Description = "Set a user secrets ID to enable secret storage";
+            command.HelpOption();
+
+            command.OnExecute(() =>
+            {
+                options.Command = new InitCommand(options.Id, options.Project);
+            });
+        }
+
+        public InitCommand(string id, string project)
+        {
+            OverrideId = id;
+            ProjectPath = project;
+        }
+
+        private static string ResolveProjectPath(string name, string path)
+        {
+            var finder = new MsBuildProjectFinder(path);
+            return finder.FindMsBuildProject(name);
+        }
+
+        private static bool HasConditionalParent(XElement el)
+        {
+            return el.Ancestors().Any(ancestor =>
+                ancestor.Attributes().Any(attribute =>
+                    attribute.Name == "Condition"));
+        }
+
+        public void Execute(CommandContext context, string workingDirectory)
+        {
+            WorkingDirectory = workingDirectory;
+            Execute(context);
+        }
+
+        public void Execute(CommandContext context)
+        {
+            var projectPath = ResolveProjectPath(ProjectPath, WorkingDirectory);
+
+            // Load the project file as XML
+            var projectDocument = XDocument.Load(projectPath);
+
+            // Confirm a UserSecretsId isn't already set
+            if (projectDocument.XPathSelectElements("//UserSecretsId").Any())
+            {
+                // TODO: i18n
+                context.Reporter.Error($"MSBuild project '{projectPath}' already contains a UserSecretsId.");
+                // TODO: correct error reporting approach
+                throw new NotImplementedException();
```

Comment:
IMO tools should not fail if there action would be a no-op. They can just exit early. I think this code should be tweaked a little so we get the following behaviors.

Was `--id` given?  | Project contains `<UserSecretsId>` already? | Action
---|---|--
No | Yes | Message: "Project is already initialized", exit code 0
No | Yes | Generate GUID, add `<UserSecretsId>` element, exit code 0
Yes | No | Add `<UserSecretsId>` element, exit code 0
Yes | Yes | Overwrite `<UserSecretsId>` element, exit code 0

Reasons to fail:
* the ID specified with `--id` is invalid.
* Project file could not be found
* Project file is malformed (not proper XML)